### ISSUE TITLE
Since 23.5, pagination is not done anymore with MergeRequests.all 

### ISSUE THREAD

**Issue by mickaeltr** [jdalrymple/gitbeaker#1221]
**Description**

When upgrading from 23.4.2 to a latter version, I noticed that pagination is not done anymore with `MergeRequests.all` (only the first page is fetched).

**Steps to reproduce**

```javascript
await gitlab.MergeRequests.all({
  scope: "all",
  sort: "asc",
  state: "opened"
})
```

**Expected behaviour**

Multiple `merge_requests` GET requests should be made.

**Actual behaviour**

A single `merge_requests` GET request is made (although there are more merge requests to fetch)

**Possible fixes**

I don't know yet, we should look at the diff between 23.4.2 and 23.5.0

**Comment by jdalrymple** [comment]
When investigating this, it seems to work fine when i dont use the scope=all query param. I also receive similar results when performing the request through insomnia, so it doesnt look like a library problem. 

![example](https://user-images.githubusercontent.com/3743662/95677549-85229b80-0bc6-11eb-89aa-837eaa43a6af.png)

**Comment by mickaeltr** [comment]
It really looks like a gitbreaker regression to me, because I only changed the version and refreshed:
<img width="998" alt="Screenshot 2020-10-11 at 14 52 46" src="https://user-images.githubusercontent.com/378910/95679394-d46ec900-0bd2-11eb-806b-d5a28650ec4a.png">

I'll try to investigate deeper later.

**Comment by jdalrymple** [comment]
I hear you. Until i can create it, i cant do much :/ When performing the request in curl, i still get a 500 error when i use the scope query parameter set to all

**Comment by jdalrymple** [comment]
https://gitlab.com/gitlab-org/gitlab-foss/-/issues/47700 This happens for me

**Comment by williamfligor** [comment]
I believe the problem is in the KyRequestor

```ts
if (!response.headers.entries().length) {
    return {};
}
```

response.headers.entries() appears to be an Interator, it doesn't have a ".length" so this condition is always false, and the headers are never returned from the responseHeadersAsObject function. Pagination depends on the headers, so this fails.

**Comment by jdalrymple** [comment]
Nice find!!

**Comment by jdalrymple** [comment]
BAM, PR pushed.

**Comment by mickaeltr** [comment]
Thanks @jdalrymple and @williamfligor

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`25.2.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/25.2.0) :rocket:
